lib/rodauth/rails/feature/internal_request.rb

```diff
@@ -0,0 +1,45 @@
+module Rodauth
+  module Rails
+    module Feature
+      module InternalRequest
+        def post_configure
+          super
+          return unless internal_request?
+
+          self.class.define_singleton_method(:internal_request) do |route, opts = {}|
+            url_options = ::Rails.application.config.action_mailer.default_url_options
+
+            scheme = url_options[:protocol]
+            port   = url_options[:port]
+            port ||= Rack::Request::DEFAULT_PORTS[scheme] if Rack.release < "2"
+            host   = url_options[:host]
+            host  += ":#{port}" if host && port
+
+            opts[:env] ||= {}
+            opts[:env]["HTTP_HOST"] ||= host if host
+            opts[:env]["rack.url_scheme"] ||= scheme if scheme
```

Comment:
This should probably also set `SERVER_NAME` and `SERVER_PORT` as Rodauth does and without overriding it we get a mix of values:

```rb
{
  "REQUEST_METHOD"=>"POST",
  "PATH_INFO"=>"/",
  "SCRIPT_NAME"=>"",
  "HTTP_HOST"=>"localhost:3000",
  "SERVER_NAME"=>"invalidurl @@.com",
  "SERVER_PORT"=>443,
  "CONTENT_TYPE"=>"application/x-www-form-urlencoded",
  "rack.input"=>#<StringIO:0x00007fc793579b50>,
  "rack.url_scheme"=>"http",
}
```

Reply:
Makes sense, added ✅